source/MRMesh/MRMeshLoadObj.cpp

```diff
@@ -2,6 +2,71 @@
 #include "MRStringConvert.h"
 #include "MRMeshBuilder.h"
 #include "MRTimer.h"
+#include "MRBuffer.h"
+#include "MRPch/MRTBB.h"
+
+#include <boost/spirit/home/x3.hpp>
+
+namespace
+{
+    using namespace MR;
+
+    tl::expected<Vector3f, std::string> parseObjVertex( const std::string_view& str )
+    {
+        using namespace boost::spirit::x3;
+
+        Vector3f v;
+        int i = 0;
+        auto coord = [&] ( auto& ctx ) { v[i++] = _attr( ctx ); };
+
+        bool r = phrase_parse(
+            str.begin(),
+            str.end(),
+            ( 'v' >> float_[coord] >> float_[coord] >> float_[coord] ),
+            ascii::space
+        );
+        if ( !r )
+            return tl::make_unexpected( "Failed to parse vertex in OBJ-file" );
+
+        return v;
+    }
+
+    struct ObjFace
+    {
+        std::vector<int> vertices;
+        std::vector<int> textures;
+        std::vector<int> normals;
+    };
+
+    tl::expected<ObjFace, std::string> parseObjFace( const std::string_view& str )
+    {
+        using namespace boost::spirit::x3;
+
+        ObjFace f;
```

Comment:
This function will allocate 3 vectors inside ObjFace every time it is called, which is slow. Better to avoid it.

Reply:
Fixed.